Formula/crackpkcs.rb

```diff
@@ -0,0 +1,23 @@
+class Crackpkcs < Formula
+  desc "Multithreaded program to crack PKCS#12 files"
+  homepage "https://crackpkcs12.sourceforge.io"
+  url "https://download.sourceforge.net/projects/crackpkcs12/files/0.2.11/crackpkcs12-0.2.11.tar.gz"
+  sha256 "9cfd0aa1160545810404fff60234c7b6372ce7fcf9df392a7944366cae3fbf25"
+  license "GPL-3.0-or-later"
+
+  depends_on "openssl@1.1"
+
+  def install
+    system "./configure",
+            *std_configure_args,
+            "--disable-silent-rules",
+            "--with-openssl=#{Formula["openssl@1.1"].opt_prefix}"
+    system "make", "install"
+  end
+
+  test do
+    # the tool doesn't have a "--version" or "--help" option
+    # However, the usage exit code is 100.
+    system "#{bin}/crackpkcs12 || [ $? -eq 100 ]"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Well, then please tell me what to test. There are some other recipes that just check the usage.

The program tries to guess the password of PKCS#12 files. The source tarball comes with some test files, but I have no access to these files in the `test` task, right?

Reply:
No, but you could either download a vulnerable file (if that doesn't take too long). Or you can create an empty file and check if the software fails as expected.

Reply:
Thanks.